src/components/FileUploadModal/index.tsx

```diff
@@ -51,6 +56,21 @@ const FileUploadModal: React.FC<FileUploadModalProps> = ({
         setIsModalVisible(false);
     };
 
+    useEffect(() => {
+        if (!fileIsDraggedOverViewer) return;
+
+        const handleWindowDrop = () => {
+            closeModal();
+        };
+
+        window.addEventListener("drop", handleWindowDrop);
+        return () => window.removeEventListener("drop", handleWindowDrop);
```

Comment:
optional: no need to wrap `closeModal` in another function here

```suggestion
        window.addEventListener("drop", closeModal);
        return () => window.removeEventListener("drop", closeModal);
```